### PR TITLE
Replace (deprecated) optparse with argparse in fits scripts

### PR DESCRIPTION
**Pull request by MSeifert04**
The [`optparse` module](https://docs.python.org/library/optparse.html) has been deprecated for a while, so I thought it might be useful to transition these scripts to use the `argparse` module.

The functionality of the scripts should not differ (the automated tests are probably not exhaustive...), however the help text for the `fitsdiff` will now look slightly different. Not sure how relevant that is.

Also I'm not sure if that needs a changelog.

### CODE REVIEW

File: astropy/io/fits/scripts/fitscheck.py [hunk 1]
```diff
@@ -40,72 +40,77 @@
...
-import textwrap
 
 from astropy.tests.helper import catch_warnings
 from astropy.io import fits
 
 
 log = logging.getLogger('fitscheck')
 
+_DESCRIPTION = """
+.e.g. fitscheck example.fits
```
Comment (pllim) [inline astropy/io/fits/scripts/fitscheck.py]: What was the leading period for? Seems like bad grammar. Did I miss something?
```suggestion
e.g. fitscheck example.fits
```

Comment (MSeifert04) [reply astropy/io/fits/scripts/fitscheck.py]: Hm... 

maybe to make it more symmetric? 🤔

Next time I'll be more careful when copy&pasting.

Comment (MSeifert04) [reply astropy/io/fits/scripts/fitscheck.py]: I'll remove that once I know if I should add a changelog, okay?

Comment (pllim) [reply astropy/io/fits/scripts/fitscheck.py]: I am leaning on no need for change log but I'll let Simon chime in. Sure, we can wait. And I don't think it is your fault, as the leading period was already there before but I just never noticed it until the diff highlighted it. :sweat_smile:

**Comment by saimn** [comment]
Thanks for taking care of this @MSeifert04, optparse will probably never be removed but it is a good idea to make the scripts more consistent and use only one method to parse arguments.
About the changelog entry, it is probably a good idea to have one as the help strings will change a bit, so it allows to track when the change happen.

File: astropy/io/fits/scripts/fitscheck.py (hunk 1 above)
Comment (saimn) [reply astropy/io/fits/scripts/fitscheck.py]: I would completely remove this sentence, it is useless as it is redundant with the help formatted by optparse/argparse. This is for example the current one with optparse:
```
Usage: fitscheck [options] <.fits files...>

        .e.g. fitscheck example.fits

        Verifies and optionally re-writes the CHECKSUM and DATASUM keywords
        for a .fits file.
        Optionally detects and fixes FITS standard compliance problems.
```

Comment (MSeifert04) [reply astropy/io/fits/scripts/fitscheck.py]: The output (especially the usage) with this PR is somewhat different:

```
$ fitscheck -h
usage: fitscheck [-h] [-k {standard,remove,none}] [-w] [-f] [-c] [-i] [-v]
                 file [file ...]

.e.g. fitscheck example.fits

Verifies and optionally re-writes the CHECKSUM and DATASUM keywords
for a .fits file.
Optionally detects and fixes FITS standard compliance problems.

positional arguments:
  file                  .fits files to process.

optional arguments:
  -h, --help            show this help message and exit
  -k {standard,remove,none}, --checksum {standard,remove,none}
                        Choose FITS checksum mode or none. Defaults standard.
  -w, --write           Write out file checksums and/or FITS compliance fixes.
  -f, --force           Do file update even if original checksum was bad.
  -c, --compliance      Do FITS compliance checking; fix if possible.
  -i, --ignore-missing  Ignore missing checksums.
  -v, --verbose         Generate extra output.
```

**Comment by MSeifert04** [comment]
I added the changelog.

**Comment by pllim** [comment]
@saimn , please approve if you are happy with this now. Thank you both!

**Comment by saimn** [comment]
Looks good, thanks Michael!

**Pull request merged**